includes/wp-cli.php

```diff
@@ -163,15 +165,252 @@ function import_from_csv( $args, $assoc_args ) {
 				WPCOM_Legacy_Redirector::insert_legacy_redirect( $redirect_from, $redirect_to );
 
 				if ( 0 == $row % 100 ) {
-					if ( function_exists( 'stop_the_insanity' ) )
+					if ( function_exists( 'stop_the_insanity' ) ) {
 						stop_the_insanity();
+					}
 					sleep( 1 );
 				}
 			}
 			fclose( $handle );
 		}
 	}
 
+	/**
+	 * Verify our redirects.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--status=<status>]:
+	 * : Filter by verification status.
+	 * ---
+	 * default: unverified
+	 * options:
+	 *   - unverified
+	 *   - verified
+	 *   - all
+	 * ---
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: csv
+	 * options:
+	 *   - table
+	 *   - json
+	 *   - yaml
+	 *   - csv
+	 * ---
+	 *
+	 * [--verbose]
+	 * : Print notifications to the console for passing URLs.
+	 *
+	 * [--strict]
+	 * : Enable verification for validated redirects to post ids.
+	 *
+	 * ## EXAMPLES
+	 *
+	 * wp wpcom-legacy-redirector verify-redirects
+	 *
+	 * @subcommand verify-redirects
+	 * @synopsis [--status=<status>] [--format=<format>] [--verbose] [--strict]
+	 */
+	function verify_redirects( $args, $assoc_args ) {
+
+		global $wpdb;
+		$post_types = get_post_types( array( 'public' => true ) );
+
+		$status = \WP_CLI\Utils\get_flag_value( $assoc_args, 'status' );
+		switch ( $status ) {
+			case 'unverified':
```

Comment:
I'm not clear on what the expected use for the verification flag is. Is it to avoid activating known broken redirects or something else?

Reply:
The default is to only process redirects that have not previously passed verification (unverified). The status flag allows this to be overridden.

- The default behavior will save a lot of time in the event of multiple batches of redirects.
- If the permalink structure changed on the site for any reason, re-verifying all redirects could be necessary.
- I don't see a huge use case for only reverifying previously verified URLs, but it seemed logical to include as an option here and required minimal overhead.

Reply:
Your mention of "activating known broken redirects" brings up an interesting point. Verification status doesn't currently have any effect over whether a redirect is used, but that could be something interesting to explore - especially if we moved closer to validation vs. verification.